src/python/pants/backend/python/packaging/pyoxidizer/target_types.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from textwrap import dedent
+
+from pants.engine.target import (
+    COMMON_TARGET_FIELDS,
+    Dependencies,
+    SingleSourceField,
+    StringField,
+    StringSequenceField,
+    Target,
+)
+
+# TODO: This runs into https://github.com/pantsbuild/pants/issues/13587
+# class PyOxidizerEntryPointField(PexEntryPointField):
+#     pass
+
+
+class PyOxidizerEntryPointField(StringField):
+    alias = "entry_point"
```

Comment:
`python_distribution` supports setting entrypoint(s) as well, IIRC. Not sure how complex it would be to default to that (...and for it to be what decides that we fallback to a `repl`, if that's going to happen).